src/wallet/abstract-wallet.ts

```diff
@@ -956,14 +946,16 @@ abstract class AbstractWallet extends EventEmitter {
       await Promise.all(
         keySplits.map(async (keySplit) => {
           const data = (await this.db.get(keySplit)) as BytesData;
-          const storedSendCommitment = msgpack.decode(arrayify(data)) as StoredSendCommitment;
+          const storedSendCommitment = msgpack.decode(
+            ByteUtils.arrayify(data),
+          ) as StoredSendCommitment;
 
           if (storedSendCommitment.txidVersion !== txidVersion) {
             return undefined;
           }
 
-          const tree = numberify(keySplit[3]).toNumber();
-          const position = numberify(keySplit[4]).toNumber();
+          const tree = parseInt(keySplit[3], 16);
```

Comment:
Is this a 1:1 replacement? the numberify function looks a bit more complex inside it. what potential values could be in the keySplit[]?

Reply:
This is a great question.

I found that keySplit[3] and [4] are 32 bytes encoded in hex, so as a string it's pretty long. But I think it's a huge overkill, because:

1. "Tree" refers to the tree number, which is just 1 (actually 0, since it starts with 0) and over time this number will grow very slowly. I'd be surprised the day it reaches 10
2. "Position" is at most 65536
3. Previously, they were converted to "BigNumbers" with numberify, but then converted to normal (i.e. small) numbers with "toNumber"

So it's technically possible for a huge 32 bytes hex to not parse well as an int, but not in practice.